src/pages/Admin/AdminUpdateClass/ClassComponent.tsx

```diff
@@ -16,41 +14,66 @@ import {
+import { ClassComponentData } from '../../../api/ClassComponentData';
+
+interface ClassComponentData {
+  id: string;
+  date: string;
+  title: string;
+  classDescription: string;
+  content?: [];
+}
 
 const ClassComponent = () => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [currentId, setCurrentID] = useState<number>();
+
   return (
     <StyledClassContainer>
-      <EditClassIcon />
-      <StyledAccordionWrapper>
-        {/* accordion header */}
-        <StyledClassHeaderWrapper>
-          <article>
-            <ClassDate>March,10th,2022</ClassDate>
-            <ClassTitleWrapper>
-              <ClassTitle>Class 1: Introduction</ClassTitle>
-              <TrashIcon />
-            </ClassTitleWrapper>
-            <ClassDescription>
-              In this course, you will learn about JavaScript data types,
-              built-in methods, and variables.
-            </ClassDescription>
-          </article>
-          <MdExpandMore />
-        </StyledClassHeaderWrapper>
+      {ClassComponentData.map((item, idx) => (
+        <div key={item.id}>
+          <EditClassIcon />
+          <StyledAccordionWrapper
+            onClick={() => {
+              setCurrentID(currentId === idx ? -1 : idx);
+              setIsExpanded(!isExpanded);
+            }}
+          >
+            {/* accordion header */}
+            <StyledClassHeaderWrapper>
+              <article>
+                <ClassDate>{item.date}</ClassDate>
+                <ClassTitleWrapper>
+                  <ClassTitle>{item.title}</ClassTitle>
+                  <TrashIcon />
+                </ClassTitleWrapper>
+                <ClassDescription>{item.classDescription}</ClassDescription>
+              </article>
+              <MdExpandMore
+                className={
+                  isExpanded && currentId === idx ? 'closed' : 'expanded'
+                }
+              />
+            </StyledClassHeaderWrapper>
+
+            {/* accordion content */}
 
-        {/* accordion content */}
-        <StyledClassItemsContainer>
-          <StyledClassItem>
-            <EditClassContentIcon />
-            <TrashIcon />
-            <span>Stream</span>
-          </StyledClassItem>
-          <AddClassWrapper>
-            <AddClassContentIcon />
-            Add new content
-          </AddClassWrapper>
-        </StyledClassItemsContainer>
-      </StyledAccordionWrapper>
+            {item.content?.map((innerElement) => (
+              <div key={innerElement}>
+                {currentId === idx && (
+                  <StyledClassItemsContainer>
+                    <StyledClassItem>
+                      <EditClassContentIcon />
+                      <TrashIcon />
+                      <span>{innerElement}</span>
+                    </StyledClassItem>
+                  </StyledClassItemsContainer>
+                )}
+              </div>
+            ))}
+          </StyledAccordionWrapper>
```

Comment:
hmm I think you can separate the `StyledAccordionWrapper` to its own file that's reusable, the `Accordion` component (`/src/components/Accordion/Accordion.tsx`).

The file would be like this
```
const Accordion = ({ header, onClick, isExpanded, children }) => {
     return (
           <StyledAccordionWrapper onClick={onClick}>
               <StyledHeaderWrapper>
                   {header}
                   <MdExpandMore
                        className={
                            isExpanded && currentId === idx ? 'closed' : 'expanded'
                        }
                   />
               </StyledHeaderWrapper>
               <StyledContentWrapper>
                   {children}
               </StyledContentWrapper>
           </StyledAccordionWrapper>
     );
}

export default Accordion;
```

and in the ClassComponent file, you do 
```
   const handleClick = () => {
       setCurrentID(currentId === idx ? -1 : idx);
       setIsExpanded(!isExpanded);
   }

   const AccordionHeader = (
             <article>
                <ClassDate>{item.date}</ClassDate>
                <ClassTitleWrapper>
                  <ClassTitle>{item.title}</ClassTitle>
                  <TrashIcon />
                </ClassTitleWrapper>
                <ClassDescription>{item.classDescription}</ClassDescription>
              </article>
    );
   
   return (
   .....
    <Accordion
        onClick={handleClick}
        header={<AccordionHeader />}
    >
        {item.content?.map((innerElement) => (
              <div key={innerElement}>
                {currentId === idx && (
                  <StyledClassItemsContainer>
                    <StyledClassItem>
                      <EditClassContentIcon />
                      <TrashIcon />
                      <span>{innerElement}</span>
                    </StyledClassItem>
                  </StyledClassItemsContainer>
                )}
              </div>
            ))}
    </Accordion>
```

something like that, does this make sense?